frame.go

```diff
@@ -26,7 +26,7 @@ type Frame struct {
 	// ID is the CAN ID
 	ID uint32
 	// Length is the number of bytes of data in the frame.
-	Length uint8
+	Length uint16
 	// Data is the frame data.
 	Data Data
 	// IsRemote is true for remote frames.
```

Comment:
We use an ethernet protocol that can actually send messages >255 bytes in length so the `uint8` wasn't sufficient to hold the message length, which is why we changed this - open to suggestions on how you'd want to deal with this

Reply:
Your ethernet protocol must be some kind of higher-layer protocol which utilizes CAN in the link layer. In that case, I think the best option would be that you implement a new library for your higher-layer protocol which depends on our library. There, you can wrap the `Frame` type in your own structs to fit your use case. As I see it this is a pure CAN library, thus we should limit the scope to the link-layer aspects and DBC file handling.

I assume you're using the descriptor library to parse signals within your protocol, meaning your signals can be more than 255 bits wide as well. This is something we can implement in this library, since signals don't really follow a standard frame structure the same way the CAN and CAN FD protocols do. That will be a breaking change but would make sense to do if we're implementing CAN FD support.

Reply:
Correct, we actually may be able to extract the message length from the header in our ethernet protocol so we can potentially revert this back to a `uint8` - the main issue with > 8 bytes (CAN-FD or arbitrary length) is that we're no longer able to use `uint64` to pack the bits so we've resorted to using `math.big` to handle this which by nature would allows lengths of any size. We can refactor some of our decoding code to deal with a 64 byte payload, it just seems like a more limiting constraint. We see value in keeping the current `can.Data` struct and associated methods since it's very efficient for messages <= 8 bytes, and we currently use both.

Would your team be open to a meeting on Zoom, etc. to talk through some of our design choices? This may be much simpler than going back-and-forth on the comments here. @jshiv and I are both in PST which is 9 hours behind Sweden but we can work something out if that sounds good on your side.

Reply:
I think that's a great idea to talk "face-to-face" :smile:

I sent you a request on LinkedIn, let's exchange contact info there.